packages/nimble-components/src/rich-text-editor/tests/rich-text-editor.spec.ts

```diff
@@ -10,4 +42,409 @@ describe('RichTextEditor', () => {
+                async () => {
+                    await waitForUpdatesAsync();
+                    const button = pageObject.getFormattingButton(value.name)!;
+                    expect(button.checked).toBeFalse();
+
+                    const event = new KeyboardEvent('keypress', {
+                        key: keyEnter
+                    } as KeyboardEventInit);
+                    button.control.dispatchEvent(event);
+
+                    expect(button.checked).toBeTrue();
+                }
+            );
+        }
+    });
+
+    describe('keyboard shortcuts should update the checked state of the buttons', () => {
+        const focused: string[] = [];
+        const disabled: string[] = [];
+
+        for (const value of formattingButtons) {
+            const specType = getSpecTypeByNamedList(value, focused, disabled);
+            // eslint-disable-next-line @typescript-eslint/no-loop-func
+            specType(
+                `"${value.name}" button keyboard shortcut check`,
+                // eslint-disable-next-line @typescript-eslint/no-loop-func
+                async () => {
+                    await waitForUpdatesAsync();
+                    const button = pageObject.getFormattingButton(value.name)!;
+                    const editor = pageObject.getTiptapEditor();
+                    expect(button.checked).toBeFalse();
+
+                    const event = new KeyboardEvent('keydown', {
+                        key: value.shortcutKey,
+                        ctrlKey: true,
+                        shiftKey: value.shiftKey,
+                        bubbles: true,
+                        cancelable: true
+                    });
+                    editor!.dispatchEvent(event);
+
+                    expect(button.checked).toBeTrue();
+                }
+            );
+        }
+    });
+
+    describe('should not leak change event through shadow DOM for buttons', () => {
+        const focused: string[] = [];
+        const disabled: string[] = [];
+
+        for (const value of formattingButtons) {
+            const specType = getSpecTypeByNamedList(value, focused, disabled);
+            // eslint-disable-next-line @typescript-eslint/no-loop-func
+            specType(
+                `"${value.name}" button not propagate change event to parent element`,
+                // eslint-disable-next-line @typescript-eslint/no-loop-func
+                () => {
+                    const button = pageObject.getFormattingButton(value.name)!;
+                    const buttonParent = button.parentElement;
+                    let parentChangeEventFired = false;
+
+                    buttonParent?.addEventListener('change', () => {
+                        parentChangeEventFired = true;
+                    });
+
+                    const event = new Event('change', { bubbles: true });
+                    button.dispatchEvent(event);
+
+                    expect(parentChangeEventFired).toBeFalse();
```

Comment:
nit: for this kind of event handling prefer to use jasmine spies. It's cleaner and gives better assertions than asserting a boolean is true. See discussion with example: https://github.com/ni/javascript-styleguide/issues/125#issuecomment-1638684399